packages/x-charts/src/ChartContainer/usePluginsMerge.ts

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { ChartsPluginTypes, ColorProcessorsConfig } from '../models';
+import { ChartSeriesType } from '../models/seriesType/config';
+import { ExtremumGettersConfig } from '../context/CartesianContextProvider';
+import { SeriesFormatterConfig } from '../context/SeriesContextProvider';
+import { defaultPlugins } from './defaultPlugins';
+
+export function usePluginsMerge<T extends ChartSeriesType>(plugins?: ChartsPluginTypes<T>[]) {
+  const defaultizedPlugins = plugins ?? defaultPlugins;
```

Comment:
I don't see the goal of this function 🤔 

Why are we overriding the given plugins? If the user overrides, to they also need to pass in the `defaultPlugins`?

Reply:
For me we have 3 usecases:

1. They do basics charts: they don't care about plugins, it works out of the box
2. They do specific chart. All they need is to pass the plugin of the chart they are building (heatmap, radar, treeview, ...)
3. They want to overrid an existing plugin (very advanced usecase) at that level of customisation, they are talented enough to pick exactly what they want from the codebase